extensions/amp-story/1.0/amp-story-page.js

```diff
@@ -1785,6 +1833,90 @@ export class AmpStoryPage extends AMP.BaseElement {
     }
   }
 
+  /**
+   * Renders the open attachment UI affordance for outlinks.
+   * @private
+   */
+  renderOpenOutlinkAttachmentUI_() {
+    const attachmentEl = this.element.querySelector(
+      'amp-story-page-attachment'
+    );
+    if (!attachmentEl) {
+      return;
+    }
+    const attachmentHref = attachmentEl.getAttribute('href');
+    if (!attachmentHref) {
+      return;
+    }
+
+    if (!this.openAttachmentEl_) {
+      this.openAttachmentEl_ = buildOpenOutlinkAttachmentElement(this.element);
+      this.openAttachmentEl_.setAttribute('href', attachmentHref);
+      this.openAttachmentEl_.addEventListener('click', () =>
+        this.openAttachment()
+      );
+
+      const textEl = this.openAttachmentEl_.querySelector(
+        '.i-amphtml-story-page-attachment-outlink-label'
+      );
+      const ctaImgEl = this.openAttachmentEl_.querySelector(
+        '.i-amphtml-story-page-attachment-outlink-image'
+      );
+
+      const openLabelAttr = attachmentEl.getAttribute('cta-text');
+      const openLabel =
+        (openLabelAttr && openLabelAttr.trim()) ||
+        getLocalizationService(this.element).getLocalizedString(
+          LocalizedStringId.AMP_STORY_PAGE_ATTACHMENT_OPEN_LABEL
+        );
+
+      const theme = attachmentEl.getAttribute('theme');
+      if (theme && AttachmentTheme.DARK === theme.toLowerCase()) {
+        this.mutateElement(() =>
+          this.openAttachmentEl_.setAttribute('theme', 'dark')
+        );
+      }
+
+      const ctaAccentColor = getStyle(attachmentEl, '--cta-accent-color');
+        if (ctaAccentColor) {
+          this.mutateElement(() =>
+            setImportantStyles(this.openAttachmentEl_, {
+              '--cta-accent-color': ctaAccentColor,
+            })
+          );
+        }
+
+      this.mutateElement(() => {
+        const ctaImgAttr = attachmentEl.getAttribute('cta-img');
+        if (ctaImgAttr !== 'none') {
+          setImportantStyles(textEl, {
+            'padding-left': '10px',
+          });
+          if (!ctaImgAttr) {
+            if (theme && AttachmentTheme.DARK === theme) {
```

Comment:
I think we should revisit how this works.

I think it's problematic to pair the accent color with white.  There are many colors that just don't pair well with white (e.g. [`#FFE6EE`](https://contrast-ratio.com/#%23FFE6EE-on-white)).

We should, instead, use [`getTextColorForRGB`](https://github.com/ampproject/amphtml/blob/a69f1d6c5df1d69151987dd91d4bedc23fc3aa70/extensions/amp-story/1.0/utils.js#L144-L175), which will tell you whether `black` or `white` will pair better with the specified accent color.